### DISCUSSION TITLE
Map distribution by torrent; local peer map sharing?

### DISCUSSION THREAD

**Discussion by myself248** [organicmaps/organicmaps#1117] [Ideas]
Downloading maps can be very slow, I'm not sure why. It took me a while to realize the circle-X icon changes into a progress bar when the download actually starts! I don't know where the maps are coming from, but I'd like to propose something like bittorrent in addition or instead:

This way, I could just add the whole world worth of torrents to my NAS seedbox, and seed them to the internet forever. And when I go to fetch a map with my phone, since I'm on the same wifi as the NAS, they'd come over at local speed. (And since I'm often wont to blow away my phone and reinstall everything, I'd feel bad about hitting the central map servers over and over for this. Hitting a torrent swarm in which I run a seed wouldn't bother me at all.) 

Furthermore, given F-Droid's ability to share APKs between phones without the internet playing a role, it'd be awesome if installed maps could be shared with other phones on the same local wifi as well. Say I'm out in the middle of nowhere and my F-Droid-using friend installs OrganicMaps from my phone, wouldn't it be cool if my installation of OM could share my maps with them too?

**Comment by biodranik** [comment]
Torrents have legal issues in other stores, and many network operators shape or even block the traffic.

The idea of sharing the map locally without the Internet is good, and it doesn't need a torrent library. It may have a limited use-case though, only between FDroid users.
